jdaviz/configs/cubeviz/plugins/tests/test_parsers.py

```diff
@@ -42,14 +42,19 @@ def test_fits_image_hdu_with_microns(image_cube_hdu_obj_microns, cubeviz_helper)
         assert cubeviz_helper.app.data_collection[i].meta[PRIHDR_KEY]['BITPIX'] == 8
 
     flux_viewer = cubeviz_helper.app.get_viewer('flux-viewer')
-    flux_viewer.on_mouse_or_key_event({'event': 'mousemove', 'domain': {'x': 0, 'y': 0}})
-    assert flux_viewer.label_mouseover.pixel == 'x=00.0 y=00.0'
-    assert flux_viewer.label_mouseover.value == '+1.00000e+00 1e-17 erg / (Angstrom cm2 s)'
+    label_mouseover = cubeviz_helper.app.session.application._tools['g-coords-info']
+    label_mouseover._viewer_mouse_event(flux_viewer,
+                                        {'event': 'mousemove', 'domain': {'x': 0, 'y': 0}})
+    assert label_mouseover.as_text() == ('Pixel x=00.0 y=00.0 Value +1.00000e+00 1e-17 erg / (Angstrom cm2 s)',  # noqa
+                                         'World 13h41m45.5759s +27d00m12.3044s (ICRS)',
+                                         '205.4398995981 27.0034178810 (deg)')  # noqa
 
     unc_viewer = cubeviz_helper.app.get_viewer('uncert-viewer')
-    unc_viewer.on_mouse_or_key_event({'event': 'mousemove', 'domain': {'x': -1, 'y': 0}})
-    assert unc_viewer.label_mouseover.pixel == 'x=-1.0 y=00.0'
-    assert unc_viewer.label_mouseover.value == ''  # Out of bounds
+    label_mouseover._viewer_mouse_event(unc_viewer,
+                                        {'event': 'mousemove', 'domain': {'x': -1, 'y': 0}})
+    assert label_mouseover.as_text() == ('Pixel x=-1.0 y=00.0',  # Out of bounds
+                                         'World 13h41m45.5759s +27d00m12.3044s (ICRS)',
+                                         '205.4398995981 27.0034178810 (deg)')  # noqa
```

Comment:
This RA and Dec should have changed because X is now -1. Only the value should not shown. The old test does not check for sky coordinates so I cannot tell you what it is supposed to be but you should double check. This does not look right.

Reply:
A simple way to check is to create that same WCS object locally and then call its `pixel_to_world` method as per https://github.com/astropy/astropy-APEs/blob/main/APE14.rst#high-level-api

Reply:
I went back to main and added debugging print statements to the original test and they gave these same values (that's not to say that they're necessarily correct).  I'll need to think about the exact scenario here to see what would be expected, or I guess we could always file as a separate follow-up and only include `as_text()[0]` in the test coverage so that we don't suggest these values are validated as correct.

Reply:
Hmm... But theoretically, this is what should happen. I wonder if I omitted the sky coordinates on purpose or this was an oversight. Past Self ™️ strikes again!

```python
from astropy.wcs import WCS
w = WCS({
        'WCSAXES': 3, 'CRPIX1': 38.0, 'CRPIX2': 38.0, 'CRPIX3': 1.0,
        'CRVAL1': 205.4384, 'CRVAL2': 27.004754, 'CRVAL3': 4.890499866509344,
        'CTYPE1': 'RA---TAN', 'CTYPE2': 'DEC--TAN', 'CTYPE3': 'WAVE',
        'CUNIT1': 'deg', 'CUNIT2': 'deg', 'CUNIT3': 'um',
        'CDELT1': 3.61111097865634E-05, 'CDELT2': 3.61111097865634E-05, 'CDELT3': 0.001000000047497451,  # noqa
        'PC1_1 ': -1.0, 'PC1_2 ': 0.0, 'PC1_3 ': 0,
        'PC2_1 ': 0.0, 'PC2_2 ': 1.0, 'PC2_3 ': 0,
        'PC3_1 ': 0, 'PC3_2 ': 0, 'PC3_3 ': 1,
        'DISPAXIS': 2, 'VELOSYS': -2538.02,
        'SPECSYS': 'BARYCENT', 'RADESYS': 'ICRS', 'EQUINOX': 2000.0,
        'LONPOLE': 180.0, 'LATPOLE': 27.004754,
        'MJDREFI': 0.0, 'MJDREFF': 0.0, 'DATE-OBS': '2014-03-30'})
```
```python
>>> w.pixel_to_world(0, 0, 0)
[<SkyCoord (ICRS): (ra, dec) in deg
     (205.4398996, 27.00341788)>,
 <SpectralCoord
    (target: <ICRS Coordinate: (ra, dec, distance) in (deg, deg, kpc)
                 (205.4384, 27.004754, 1000.)
              (pm_ra_cosdec, pm_dec, radial_velocity) in (mas / yr, mas / yr, km / s)
                 (0., 0., 0.)>)
   4.89049987e-06 m>]
>>> w.pixel_to_world(-1, 0, 0)
[<SkyCoord (ICRS): (ra, dec) in deg
     (205.43994013, 27.00341788)>,
 <SpectralCoord
    (target: <ICRS Coordinate: (ra, dec, distance) in (deg, deg, kpc)
                 (205.4384, 27.004754, 1000.)
              (pm_ra_cosdec, pm_dec, radial_velocity) in (mas / yr, mas / yr, km / s)
                 (0., 0., 0.)>)
   4.89049987e-06 m>]
```

Reply:
So it turns out when you load that test case into Cubeviz, the flux cube WCS becomes a `PaddedSpectrumWCS` but the uncert cube retained the original FITS WCS. If you do this, you can recreate the weird results this test is seeing.

```python
>>> cubeviz_helper.app.data_collection[1].coords.pixel_to_world(0, 0, 3)
[<SpectralCoord 
    (target: <ICRS Coordinate: (ra, dec, distance) in (deg, deg, kpc)
                 (205.4384, 27.004754, 1000.)
              (pm_ra_cosdec, pm_dec, radial_velocity) in (mas / yr, mas / yr, km / s)
                 (0., 0., 0.)>)
   4.89049987e-06 m>,
 <SkyCoord (ICRS): (ra, dec) in deg
     (205.43977801, 27.00341788)>]
>>> cubeviz_helper.app.data_collection[1].coords.pixel_to_world(-1, 0, 3)
[<SpectralCoord 
    (target: <ICRS Coordinate: (ra, dec, distance) in (deg, deg, kpc)
                 (205.4384, 27.004754, 1000.)
              (pm_ra_cosdec, pm_dec, radial_velocity) in (mas / yr, mas / yr, km / s)
                 (0., 0., 0.)>)
   4.89149987e-06 m>,
 <SkyCoord (ICRS): (ra, dec) in deg
     (205.43977801, 27.00341788)>]
```

It makes no sense but (1) it is unrelated to this PR, and (2) it is not real data. So, my recommendation is that you retain the old test behavior by not comparing the sky coordinates at all. I'd rather we do that than to say in the test we expect non-sensible result.

I thought uncert would have the same WCS as flux cube... maybe @bmorris3 cube work would naturally fix this. 🤷

Reply:
> So, my recommendation is that you retain the old test behavior by not comparing the sky coordinates at all.

ok, good idea, I like that plan for now.